services/gps-exporter/src/main.rs

```diff
@@ -0,0 +1,336 @@
+        );
+        from = max_date;
+    }
+
+    println!("🎉 {}", Green.bold().paint("Exporting GPS"));
+    println!("From: {}", White.bold().paint(from.to_rfc2822()));
+    println!("To: {}", White.bold().paint(to.to_rfc2822()));
+    println!(
+        "File span: {} seconds\n",
+        White.bold().paint(segment_length.to_string())
+    );
+
+    for t in (from.timestamp()..to.timestamp()).step_by(segment_length as usize) {
+        let start = Utc.from_utc_datetime(&NaiveDateTime::from_timestamp(t, 0));
+        let stop = start + Duration::seconds(segment_length as i64);
+        let filename = out_dir
+            .join(Path::new(&start.to_rfc3339()))
+            .with_extension("gpx");
+
+        print!(
+            "{} ",
+            ANSIStrings(&[
+                Blue.paint("Writing segment: "),
+                Blue.bold().paint(
+                    filename
+                        .file_name()
+                        .unwrap_or_default()
+                        .to_str()
+                        .unwrap_or_default()
+                )
+            ])
+        );
+
+        let recs: Vec<GPS> = sqlx::query_as(
+            r#"
+            SELECT time, lat, lng
+            FROM gps
+            WHERE time BETWEEN $1 AND $2
+            ORDER BY time
+            "#,
+        )
+        .bind(&start)
+        .bind(&stop)
+        .fetch_all(&pool)
+        .await?;
+
+        if recs.len() == 0 {
+            println!("{}", Yellow.paint("⇝ No data"));
+            continue;
+        }
+
+        let mut segment = TrackSegment::new();
+        for rec in recs {
+            let mut wpt = Waypoint::new(Point::new(rec.lng, rec.lat));
+            wpt.time = Some(rec.time);
+
+            segment.points.push(wpt);
+        }
+
+        let mut track = Track::new();
+        track.segments.push(segment);
+
+        let mut data: Gpx = Default::default();
+        data.version = GpxVersion::Gpx11;
+        data.tracks.push(track);
+
+        let path = Path::new(out_dir).join(Path::new(&start.to_rfc3339()).with_extension("gpx"));
+        let out_file = std::fs::File::create(&path)?;
+
+        match write(&data, out_file) {
+            Ok(_) => println!("{}", Green.paint("✓")),
+            Err(_) => eprintln!("{} {:?}", Red.bold().paint("⨯ "), path),
+        };
+    }
+
+    println!(
+        "\n{}\n",
+        Green.bold().paint("🥳 Export complete. Enjoy your 📊!")
+    );
+
+    Ok(())
+}
+
+fn parse_date(date: &str) -> anyhow::Result<DateTime<Utc>> {
+    Utc.datetime_from_str(date, "%Y-%m-%d %H:%M:%S")
+        .with_context(|| "Use YYYY-MM-DD HH:MM:SS")
+}
+
+fn is_date(date: String) -> Result<(), String> {
+    match parse_date(&date) {
+        Ok(_) => Ok(()),
+        Err(e) => Err(e.to_string()),
+    }
+}
+
+fn is_u64(num: String) -> Result<(), String> {
+    match num.parse::<u64>() {
+        Ok(_) => Ok(()),
+        Err(_) => Err(format!("Must be a positive integer")),
+    }
+}
+
+/*
+#[derive(Debug)]
```

Comment:
Wouldn't mind a comment here explaining that the shapefile lib is broken and a link that will let us know the status of the fix